### DISCUSSION TITLE
Capture Photo using HTML5, JS and Budibase

### DISCUSSION THREAD

**Discussion by xnohat** [Budibase/budibase#6478] [Show and tell]
Hi

I see some guys asked how to capture photo using camera and budibase.

I tried to do some hacks with javascript and budibase to make this idea work

Idea:
Capture a photo with a photo frame by Budibase 

![IMG_2230](https://user-images.githubusercontent.com/4613808/176008419-0f882313-d7a4-41e2-87ac-26cabdbb8093.JPG)

Solutions:
1. Create a html file with javascript html5 api to capture photo using camera device (ex: your phone camera), merge with a photo frame (transparent background .png)
2. Upload HTML file to Budibase (for javascript CORS policy)
3.  When javascript capture photo, encode photo to base64, sent to Budibase WebHook and save to Budibase internal database

I attach with this post source code of budibase app, html file with js capture photo, and example photo frame
Import & modify it.

[photocapture-export-1656352623565.txt](https://github.com/Budibase/budibase/files/8994645/photocapture-export-1656352623565.txt)

rename this file to jscapimage.html before upload to budibase internal database
[jscapimage.txt](https://github.com/Budibase/budibase/files/8994651/jscapimage.txt)

![birthdayframe](https://user-images.githubusercontent.com/4613808/176009490-6f4b15fb-fc25-4943-b0f9-a9df3e9a7b4a.png)

**Comment by melohagan** [comment]
Thanks for sharing! 😄 

Out of curiosity, which part does the base64 encoding of the photo or is that handled externally?
